features/features.go

```diff
@@ -220,3 +234,32 @@ func Flag(tr confGetter, feature SnapdFeature) (bool, error) {
 	}
 	return false, fmt.Errorf("%s can only be set to 'true' or 'false', got %q", feature, isEnabled)
 }
+
+type FeatureInfo struct {
+	Supported bool `json:"supported"`
+	Enabled   bool `json:"enabled"`
+}
```

Comment:
could we add a string field eg. `UnsupportedReason` which would carry information why a given feature is not supported?

Reply:
I like this idea a lot, thanks!